### PR TITLE
Show correct ERC20 token balances

### PR DESCRIPTION
**Pull request by elgatovital**
Handles #747 
Handles #692

Changes proposed in this pull request:
- Pass data to gmail outside of intent extras due to gmail's ignorance!
- Parse balances correctly
- Add formatting pattern for ERC20 balances

@gnosis/mobile-devs

### CODE REVIEW

File: app/src/main/java/io/gnosis/safe/ui/settings/app/GetInTouchFragment.kt [hunk 1]
```diff
@@ -86,7 +85,12 @@ class GetInTouchFragment : BaseViewBindingFragment<FragmentGetInTouchBinding>()
             val feedback = helper.createFeedbackText(requireContext())
             val intent = Intent(Intent.ACTION_SENDTO).apply {
                 type = "text/html"
-                data = Uri.parse("mailto:${getString(R.string.email_feedback)}")
+                // gmail is ignoring intent extras
+                data = Uri.parse(
```
Comment (DmitryBespalov) [inline app/src/main/java/io/gnosis/safe/ui/settings/app/GetInTouchFragment.kt:89]: will it have any effect on other mail clients (non-gmail), i.e. would they have duplicate subject/text or body if this URI is used and the intent extras - as the code is doing now?

File: data/src/main/java/io/gnosis/data/adapters/MoshiBuilder.kt [hunk 2]
```diff
@@ -14,8 +15,13 @@ internal val transactionDtoAdapter =
         .withDefaultValue(UnknownTransactionDto)
 
 val dataMoshi =
-    MoshiBuilderFactory.makeMoshiBuilder()
+    Moshi.Builder()
```
Comment (rmeissner) [inline data/src/main/java/io/gnosis/data/adapters/MoshiBuilder.kt:18]: Why can the factory not be used anymore? Or other question why is the order important, shouldn't the annotation make it order independent.

File: app/src/main/java/io/gnosis/safe/ui/settings/app/GetInTouchFragment.kt [hunk 3]
```diff
@@ -86,7 +85,12 @@ class GetInTouchFragment : BaseViewBindingFragment<FragmentGetInTouchBinding>()
...
+                    "mailto:${getString(R.string.email_feedback)}" +
+                            "?subject=${getString(R.string.feedback_subject)}" +
```
Comment (rmeissner) [inline app/src/main/java/io/gnosis/safe/ui/settings/app/GetInTouchFragment.kt:91]: Does this work with other email clients as well?

File: app/src/main/java/io/gnosis/safe/ui/settings/app/GetInTouchFragment.kt (hunk 1 above)
Comment (elgatovital) [reply app/src/main/java/io/gnosis/safe/ui/settings/app/GetInTouchFragment.kt:89]: it is possible that some clients use intent extras and some just data uri. In worst case data will just be overridden.

File: app/src/main/java/io/gnosis/safe/ui/settings/app/GetInTouchFragment.kt (hunk 3 above)
Comment (elgatovital) [reply app/src/main/java/io/gnosis/safe/ui/settings/app/GetInTouchFragment.kt:91]: yes

File: data/src/main/java/io/gnosis/data/adapters/MoshiBuilder.kt (hunk 2 above)
Comment (elgatovital) [reply data/src/main/java/io/gnosis/data/adapters/MoshiBuilder.kt:18]: we still use factory. But I think it is more feasible to have moshi constructed in the app itself and not in the library. Library should not make too many assumptions about how moshi will be used or what type adapters will be needed.

Comment (rmeissner) [reply data/src/main/java/io/gnosis/data/adapters/MoshiBuilder.kt:18]: Where do we use it? Shouldn't we use the same approach everywhere?

Comment (elgatovital) [reply data/src/main/java/io/gnosis/data/adapters/MoshiBuilder.kt:18]: we probably won't need that many adapters/type converters in other places. Also having a complete builder at one place is better for readability I think.

Comment (rmeissner) [reply data/src/main/java/io/gnosis/data/adapters/MoshiBuilder.kt:18]: This feels weird, if we think it is better to use then use it. In general I am more a fan of being explicit here and would remove the factory (as a lot of stuff in the moshi builder is order dependant)

Comment (elgatovital) [reply data/src/main/java/io/gnosis/data/adapters/MoshiBuilder.kt:18]: here is pr for this: https://github.com/gnosis/svalinn-kotlin/pull/49 
I would release new patch version for savlinn and update dependencies within this pr @rmeissner

**Pull request merged**